components/src/dynamo/sglang/checkpoint_restore.py

```
@@ -0,0 +1,249 @@
+class SGLangCheckpointAdapter:
+    """Adapts an sgl.Engine to the sleep/wake_up interface expected by
+    CheckpointConfig.run_lifecycle (matching vLLM's AsyncLLM API).
+
+    sleep():   pause generation -> release GPU memory
+    wake_up(): resume GPU memory -> continue generation
+    """
+
+    def __init__(self, engine: sgl.Engine):
+        self._engine = engine
+
+    async def sleep(self, level: int = 1) -> None:
```

Ruff (0.15.2)

[warning] Unused method argument: `level`

(ARG002)